src/Mobs/PassiveMonster.h

```diff
@@ -23,7 +24,22 @@ class cPassiveMonster :
 	/** Returns the items that the animal of this class follows when a player holds it in hand. */
 	virtual void GetFollowedItems(cItems & a_Items) { }
 
-} ;
+	/** Returns the items that make the animal breed - this is usually the same as the ones that make the animal follow, but not necessarily. */
+	virtual void GetBreedingItems(cItems & a_Items) { GetFollowedItems(a_Items); }
+
+	cPassiveMonster * GetPartner(void) const { return m_LovePartner; }
+	void EngageLoveMode(cPassiveMonster * a_Partner);
+	void ResetLoveMode();
+
+	bool IsInLove() const { return (m_LoveTimer > 0); }
+	bool IsInLoveCooldown() const { return (m_LoveCooldown > 0); }
+
+protected:
+	cPassiveMonster * m_LovePartner;
+	int m_LoveTimer;
+	int m_LoveCooldown;
+	int m_MatingTimer;
```

Comment:
These should have been documented - what they do, how they interact with each other etc.